File: content/problems/1337-the-k-weakest-rows-in-a-matrix/solutions/binary-count-pair-sort/solution.cpp
class Solution {
    public: vector<int> kWeakestRows(vector<vector<int>>& mat,int k){
        vector<pair<int,int>> rows;
        for(int r=0;r<(int)mat.size();++r){
            int soldiers=lower_bound(mat[r].begin(),mat[r].end(),0,greater<int>())-mat[r].begin();
            rows.push_back({
                soldiers,r
            });
        }
        sort(rows.begin(),rows.end());
        vector<int> answer;
        for(int i=0;i<k;++i)answer.push_back(rows[i].second);
        return answer;
    }
};

File: content/problems/1337-the-k-weakest-rows-in-a-matrix/solutions/binary-count-pair-sort/Solution.java
class Solution {
    public int[] kWeakestRows(int[][] mat,int k){
        Integer[] rows=new Integer[mat.length];
        int[] soldiers=new int[mat.length];
        for(int r=0;r<mat.length;r++){
            rows[r]=r;
            int left=0,right=mat[r].length;
            while(left<right){
                int middle=(left+right)/2;
                if(mat[r][middle]==1)left=middle+1;
                else right=middle;
            }
            soldiers[r]=left;
        }
        Arrays.sort(rows,(a,b)->soldiers[a]!=soldiers[b]?soldiers[a]-soldiers[b]:a-b);
        int[] answer=new int[k];
        for(int i=0;i<k;i++)answer[i]=rows[i];
        return answer;
    }
}
